src/jit/morph.cpp

```diff
@@ -8858,61 +8858,13 @@ GenTreePtr Compiler::fgMorphCall(GenTreeCall* call)
     // We need to do these before the arguments are morphed
     if ((call->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC))
     {
-        CorInfoIntrinsics methodID = info.compCompHnd->getIntrinsicID(call->gtCallMethHnd);
+        // See if this is foldable
+        GenTree* optTree = gtFoldExprCall(call);
 
-        if (methodID == CORINFO_INTRINSIC_Illegal)
+        // If we optimized, morph the result
+        if (optTree != call)
         {
-            // Check for a new-style jit intrinsic.
-            const NamedIntrinsic ni = lookupNamedIntrinsic(call->gtCallMethHnd);
-
-            if (ni == NI_System_Enum_HasFlag)
-            {
-                GenTree* thisOp  = call->gtCallObjp;
-                GenTree* flagOp  = call->gtCallArgs->gtOp.gtOp1;
-                GenTree* optTree = gtOptimizeEnumHasFlag(thisOp, flagOp);
-                if (optTree != nullptr)
-                {
-                    return fgMorphTree(optTree);
-                }
-            }
-        }
-
-        genTreeOps simpleOp = GT_CALL;
-        if (methodID == CORINFO_INTRINSIC_TypeEQ)
-        {
-            simpleOp = GT_EQ;
-        }
-        else if (methodID == CORINFO_INTRINSIC_TypeNEQ)
-        {
-            simpleOp = GT_NE;
-        }
-
-        if (simpleOp == GT_EQ || simpleOp == GT_NE)
-        {
-            noway_assert(call->TypeGet() == TYP_INT);
-
-            // Check for GetClassFromHandle(handle) and obj.GetType() both of which will only return RuntimeType
-            // objects. Then if either operand is one of these two calls we can simplify op_Equality/op_Inequality to
-            // GT_NE/GT_NE: One important invariance that should never change is that type equivalency is always
-            // equivalent to object identity equality for runtime type objects in reflection. This is also reflected
-            // in RuntimeTypeHandle::TypeEquals. If this invariance would ever be broken, we need to remove the
-            // optimization below.
-
-            GenTreePtr op1 = call->gtCallArgs->gtOp.gtOp1;
-            GenTreePtr op2 = call->gtCallArgs->gtOp.gtOp2->gtOp.gtOp1;
-
-            if (gtCanOptimizeTypeEquality(op1) || gtCanOptimizeTypeEquality(op2))
-            {
-                JITDUMP("Optimizing call to Type:op_%s to simple compare via %s\n",
-                        methodID == CORINFO_INTRINSIC_TypeEQ ? "Equality" : "Inequality", GenTree::OpName(simpleOp));
-
-                GenTreePtr compare = gtNewOperNode(simpleOp, TYP_INT, op1, op2);
-
-                // fgMorphSmpOp will further optimize the following patterns:
-                //  1. typeof(...) == typeof(...)
-                //  2. typeof(...) == obj.GetType()
-                return fgMorphTree(compare);
-            }
+            return fgMorphTree(optTree);
```

Comment:
Currently Github is saying that you have a merge conflict to resolve in morph.

Reply:
Indeed, will need to rebase and fix.